lightwood/api/data_source.py

```diff
@@ -345,35 +353,25 @@ def _prepare_encoders(self):
         with all available data for that column.
         """
         encoders = {}
-    
         input_encoder_training_data = {'targets': [], 'previous': []}
 
-        for config in self.config['output_features']:
-            column_name = config['name']
-            column_data = self.get_column_original_data(column_name)
-
-            encoder_instance = self._prepare_column_encoder(config, is_target=True)
-
-            input_encoder_training_data['targets'].append({
-                'encoded_output': encoder_instance.encode(column_data),
-                'unencoded_output': column_data,
-                'output_encoder': encoder_instance,
-                'output_type': config['type']
-            })
-
-            encoders[column_name] = encoder_instance
-
         previous_cols = []
         for config in self.config['input_features']:
             column_name = config['name']
             if column_name.startswith('__mdb_ts_previous_'):
                 column_data = self.get_column_original_data(column_name)
                 previous_cols.append(column_name)
-                input_encoder_training_data['previous'].append({'data': column_data,
-                                                                'name': column_name,
-                                                                'original_type': config['original_type'],
-                                                                'output_type': config['type']
-                                                                })
+                col_info = {'data': column_data,
+                            'name': column_name,
+                            'original_type': config['original_type'],
+                            'group_info': {conf['name']: self.get_column_original_data(conf['name'])
+                                           for conf in self.config['input_features'] if conf['grouped_by']},
+                            'output_type': config['type']
+                            }
+                col_info = generate_target_group_normalizers(col_info)
```

Comment:
Just to be clear on this:
* It will still get triggered when no group bys exist
* It will just generate the `__default` normalizer which will be used for everything

Or is the no-group-by situation handled some other way ?

Reply:
That's indeed what happens, correct